src/index.js

```diff
@@ -63,44 +88,155 @@ async function mountTools() {
   }
 }
 
+/*
+  Render functions
+  Differences:
+  Vue2 passes in a `createElement` function. Vue3 exports `h` to create vnodes
+  Vue2's createElement expects createElement("div", { attrs: { id: "..." }})
+  Vue3's h expects h("div", { id: "..." })
+*/
+function renderVue2(createElement) {
+  return createElement("div", [
+    createElement("div", {
+      attrs: {
+        id: `userfront-${this.toolId}`,
+      },
+    }),
+  ]);
+}
+function renderVue3() {
+  return h("div", [
+    h("div", {
+      id: `userfront-${this.toolId}`,
+    }),
+  ]);
+}
+
+// Component config
+const componentConfig = {
+  props: {
+    toolId: {
+      type: String,
+    },
+  },
+  // Call the appropriate render function
+  render(...args) {
+    if (isVue3) {
+      return renderVue3.call(this, ...args)
+    } else {
+      return renderVue2.call(this, ...args)
+    }
+  },
+  async mounted() {
+    await mountTools();
+  },
+}
+
 /**
- * Returns a compiled Vue component.
+ * Registers a Vue component for a toolkit item.
  * Uses "render" instead of "template" to avoid runtime compiler:
  * https://vuejs.org/v2/guide/installation.html#Runtime-Compiler-vs-Runtime-only
  *
- * @param {Object} name - Vue component
+ * @param {object} app - Vue app to register component on (Vue 2 singleton or Vue 3 instance)
+ * @param {string} name - name of the component
  * @returns
  */
-export function build(name) {
-  return Vue.component(name, {
-    props: {
-      toolId: {
-        type: String,
-        // required: true,
-      },
-    },
-    render(createElement) {
-      return createElement("div", [
-        createElement("div", {
-          attrs: {
-            id: `userfront-${this.toolId}`,
-          },
-        }),
-      ]);
-    },
-    async mounted() {
-      await mountTools();
-    },
-  });
+function register(app, name) {
+  if (!app.component) {
+    console.warn('@userfront/vue: tried to register a component without being passed a Vue instance. This is likely a bug with @userfront/vue.')
+    return
+  }
+  if (name) {
+    return app.component(name, componentConfig);
+  }
+  console.warn('@userfront/vue: tried to register a component without a name. This is likely a bug with @userfront/vue.')
+}
+
+/**
+ * If Vue 2 is in use, build component with the Vue singleton.
+ * If Vue 3 is in use, give a useful console error and throw.
+ * If Vue is not found, throw.
+ * @param {string} name - name of component to register with the Vue singleton
+ * @returns 
+ */
+function legacyBuild(name) {
+  // Vue 3: cannot use this method
+  // Log a helpful warning, and throw an error so client gets a stack trace
+  if (isVue3) {
```

Comment:
This only happens if someone explicitly calls `Userfront.build` in Vue 3. I'm not sure if `build` has ever been a documented way to get Vue components, but just in case...